internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -745,6 +733,21 @@ func dataSourceKubernetesCluster() *pluginsdk.Resource {
 			Computed:   true,
 			Deprecated: "This property is deprecated and will be removed in v4.0 of the AzureRM Provider in favour of the `node_public_ip_enabled` property.",
 		}
+		resource.Schema["storage_profile"].Elem.(*pluginsdk.Resource).Schema["disk_driver_version"] = &pluginsdk.Schema{
+			Deprecated: "This feature is a preview feature and will be removed in version 4.0 of the AzureRM Provider.",
```

Comment:
Please update all of the deprecation messages to the following:
`This property is not available in the stable API and will be removed in v4.0 of the Azure Provider. Please see https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#aks-migration-to-stable-api for more details`

The link in the message isn't accessible right now, but it will be when the release goes out this week.